### PR TITLE
Add rotation operator

### PR DESCRIPTION
**Pull request by Vici37**
It feels a bit more chaotic than I was hoping, and a lot of that comes from the different arguments you can pass in. Performance wise it's similar to the bilinear resize due to each pixel needing to the calculated individually.

See [documentation](http://troy.sornson.io/cr-image/CrImage/Operation/Rotate.html) for the same operated added to cr-image.

### CODE REVIEW

File: src/pluto/operation/rotate.cr [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    center_y : Int32 = height // 2,
+    radius : Int32 = -1,
+    pad : Bool = false,
+    pad_type : PaddingType = PaddingType::Black
+  ) : self
+    clone.rotate!(degrees, center_x: center_x, center_y: center_y, radius: radius, pad: pad, pad_type: pad_type)
+  end
+
+  # Can't really find good ways to reduce this more than it is, so disabling this check here
+  # ameba:disable Metrics/CyclomaticComplexity
```
Comment (sanks64) [inline src/pluto/operation/rotate.cr:15]: Please disable this globally:

```yaml
# .ameba.yml

Metrics/CyclomaticComplexity:
  Enabled: false
```

File: src/pluto/operation/rotate.cr [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  def rotate!(
+    degrees : Float64,
+    *,
+    center_x : Int32 = width // 2,
+    center_y : Int32 = height // 2,
+    radius : Int32 = -1,
+    pad : Bool = false,
```
Comment (sanks64) [inline src/pluto/operation/rotate.cr:22]: Please replace all mentions of "pad" with "padding".

**Comment by sanks64** [comment]
Should we mention this new operation in `README.md`?

**Comment by Vici37** [comment]
Doh, forgot to update the README, thanks for calling that out :P

File: src/pluto/operation/rotate.cr (hunk 2 above)
Comment (Vici37) [reply src/pluto/operation/rotate.cr:22]: Doh! Can do :+1:

File: README.md [hunk 3]
```diff
@@ -33,6 +33,7 @@
   - Gaussian blur
   - Horizontal blur
   - Padding
+  - Rotation
```
Comment (sanks64) [inline README.md:36]: This differs from the method/module name. I think it would be great to have them identical.

Comment (Vici37) [reply README.md:36]: That's a good observation! What do you think the appropriate naming convention should be for operation?

I've been typically using the verb form of words (`pad`, `rotate`) as the name of the method to _do_ stuff with. In lists like this, it reads better when it's the noun form of the word (`padding`, `rotation`), and with the desire have these two names being the same, it leads to method names taking the noun form.

I'm good renaming the method to the noun form, I just this realization now and wanted to share, to see if my own observation matches reality.

Comment (Vici37) [reply README.md:36]: Unless you think `rotate` (Rotate) is the better word to use everywhere?

Comment (sanks64) [reply README.md:36]: The noun form seems good. I don't have a strong preference, I just want it to be consistent. The verb form would also look weird in `README.md`.

File: spec/pluto/operation/rotate_spec.cr [hunk 4]
```diff
@@ -0,0 +1,55 @@
+require "../../spec_helper"
+
+describe Pluto::Operation::Rotation do
+  describe "#rotation" do
+    it "checks padding and radius" do
+      image = ga_sample
+
+      expect_raises(Exception, "Can't padding image and limit rotation by radius") do
```
Comment (sanks64) [inline spec/pluto/operation/rotate_spec.cr:8]: Sorry, I was about to merge the PR, but noticed this. I think it still need to be "pad" here?

Comment (sanks64) [reply spec/pluto/operation/rotate_spec.cr:8]: Feel free to go ahead and merge the PR after fixing this.

**Review by sanks64 — DISMISSED** [review]

**Comment by sanks64** [comment]
Thanks for the PR!

File: spec/pluto/operation/rotate_spec.cr (hunk 4 above)
Comment (Vici37) [reply spec/pluto/operation/rotate_spec.cr:8]: Ha! Good catch. Thwarted by find-and-replace.

**Pull request merged**